src/backend/ckusdc-pool/main.mo

```diff
@@ -269,4 +281,30 @@ actor CkusdcPool {
 		{ rate = usdUsdc.rateF; timestamp = usdUsdc.timestamp };
 	};
 
+	public shared ({ caller }) func weekly_reward_approval({ memo; created_at_time; amount; expires_at } : RewardApprovalArg) : async Result<Nat, RewardApprovalErr> {
+		let stakingCanister = Principal.fromText("mhahe-xqaaa-aaaag-qndha-cai");
+
+		if (caller != stakingCanister) {
+			return #err(#NotAuthorised);
+		};
+
+		let approveArg : Icrc.ApproveArgs = {
+			fee = null;
+			memo;
+			from_subaccount = null;
+			created_at_time;
+			amount;
+			expected_allowance = null;
+			expires_at;
+			spender = { owner = stakingCanister; subaccount = null }
```

Comment:
all stake depositing in subaccount null 
for fee can we choose another subaccount
```suggestion
			spender = { owner = stakingCanister; subaccount = null }
```